### PR TITLE
Add self_placed block filter

### PR DESCRIPTION
**Pull request by Jeck**
This PR adds a filter to check if a given block was placed by the player themself, not just by any player. It functions using the same basic functionality as "player_placed".

### What good is a self_placed filter?

- A skill that levels up when breaking other player's stuff, but not your own.
- Weapons that deal bonus damage when standing on home turf.
- Tools that mine your own blocks instantly.
- Explosives that don't damage your own blocks
- Probably other uses too.

### How does it work?

The implementation relies on https://github.com/Auxilor/eco/pull/243 this pull request.
TL;DR -- eco currently stores an integer "1" for any block placed by a player. The actual value stored is never checked, just its existence. I've taken advantage of that to store a value identifying the placing player instead.

### CODE REVIEW

**Comment by WillFP** [comment]
Closing this because it's incompatible with v4
